### ISSUE TITLE
Windows arguing about pyinstaller being a 16 bit application

### ISSUE THREAD

**Issue by Ampter** [pyinstaller/pyinstaller#8955]
## Description of the issue
I installed pyinstaller through pip, then i tried using it but windows says it is a 16 bit app and doesn`t do anything. Installed it yesterday, reinstalled but still not working.
### Context information (for bug reports)

* Output of `pyinstaller --version`: ```Can`t even run, output : pyinstaller --version
This version of C:\Users\user\Documents\Python Scripts\Python_weather_tools\pyinstaller.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher. first a window opens that says pyinstaller is 16 bit. same with debug and log option```
* Version of Python: <3.11.9>
* Platform:  Windows 11 Home edition 
* Installed python through python.org/downloads

**Comment by bwoodsend** [comment]
`C:\Users\user\Documents\Python Scripts\Python_weather_tools\pyinstaller.exe` is not where `pyinstaller.exe` would be installed (it should end in `\Scripts\pyinstaller.exe`). Are you sure you've not got another executable called `pyinstaller.exe` in `PATH` shadowing it?

**Comment by rokm** [comment]
Yeah, that `Python_weather_tools` in the path looks suspicious. If you copied the error message from the error dialog, can you take a screenshot instead?

**Comment by Ampter** [comment]
> `C:\Users\user\Documents\Python Scripts\Python_weather_tools\pyinstaller.exe` is not where `pyinstaller.exe` would be installed (it should end in `\Scripts\pyinstaller.exe`). Are you sure you've not got another executable called `pyinstaller.exe` in `PATH` shadowing it?

Where can i check that?

**Comment by rokm** [comment]
> Where can i check that?

Try running the following commands and paste the output for each of them:

```
where pyinstaller
where pyi-bindepend
where pip
where python
```

**Comment by Ampter** [comment]
I think i found out the issue but i am not sure

**Comment by Ampter** [comment]
sorry, the solution was that windows was looking at the Python 12 folder, but i use 3.11.9. I deleted the python 3.12 folder, reinstalled the pyinstaller module and it worked.
